pkg/pillar/types/global.go

```diff
@@ -114,6 +117,165 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota + 1
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     uint32
+	intMax     uint32
+	intDefault uint32
+
+	stringDefault   string
+	settingDefault	string
+	boolDefault     bool
+	triStateDefault TriState
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
+	agentSettings map[string]ConfigItemSpec
+}
+
+func (configMap *ConfigItemSpecMap) AddIntItem(key string, defaultInt uint32, min uint32, max uint32) {
```

Comment:
You need another parameter to add it to GlobalSettings or agentSettings

AddIntItem(key string, defaultInt uint32, min uint32, max uint32, globalSetting, agentSetting bool)

if globalSetting {
     configMap.globalSettings[key] = configItem
}
if agentSetting {
     configMap.agentSettings[key] = configItem
}